### PR TITLE
Add showcase of event handling, BLoC and routing (#31, #32, #30)

### PR DESCRIPTION
**Pull request by jowczar**
This code is taken from the tutorial: https://perpet.io/blog/how-to-build-a-clubhouse-clone-app-with-flutter-a-tutorial-by-perpetio-part-1/

There is a problem with some of the original tutorial code as some parts are deprecated, for example, mapEventToState is no longer usable:
https://github.com/felangel/bloc/issues/2526

I already changed that code so it should work now.

IMPORTANT: This is an example/showcase and only forgot password /sign-in pages are working to show flutter developers how to manage routes, what architecture we will use, how to do event handling, and what bloc providers are for.

It should be used as an example! Don't treat this as part of the target application.

### CODE REVIEW

File: frontend/lib/components/button.dart [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            ),
+            child: Row(
+              children: <Widget>[
+                Expanded(
+                  child: Text(
+                    text,
+                    textAlign: TextAlign.center,
+                    style: TextStyle(
+                      fontSize: 25,
+                    ),
```
Comment (ShanghaiG) [inline frontend/lib/components/button.dart:34]: ah yes, the ")" set :DD

File: frontend/lib/core/const/text_constants.dart [hunk 2]
```diff
@@ -0,0 +1,169 @@
+// TODO: I copied this from tutorial, unsure if it's needed – remove if we prefer to use strings directly
```
Comment (ShanghaiG) [inline frontend/lib/core/const/text_constants.dart:1]: i think that better approach is to keep strings out of main code, so keeping it here is very good idea imo

File: frontend/lib/core/service/validation_service.dart [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class ValidationService {
+  static bool username(String text) {
+    return text.length > 1;
+  }
+
+  static bool email(String text) {
+    bool emailValid = RegExp(
+            r"^[a-zA-Z0-9.a-zA-Z0-9.!#$%&'*+-/=?^_`{|}~]+@[a-zA-Z0-9]+\.[a-zA-Z]+")
```
Comment (ShanghaiG) [inline frontend/lib/core/service/validation_service.dart:8]: hmm, its acceptable here due to situation we will not use it anywhere else in the code (i think, and i hope so)

File: frontend/lib/core/service/validation_service.dart [hunk 4]
```diff
@@ -0,0 +1,20 @@
+class ValidationService {
+  static bool username(String text) {
+    return text.length > 1;
```
Comment (ShanghaiG) [inline frontend/lib/core/service/validation_service.dart:3]: im curious, could it be written as something like `Boolean(text.length)` ?

File: frontend/lib/core/service/validation_service.dart [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  }
+
+  static bool email(String text) {
+    bool emailValid = RegExp(
+            r"^[a-zA-Z0-9.a-zA-Z0-9.!#$%&'*+-/=?^_`{|}~]+@[a-zA-Z0-9]+\.[a-zA-Z]+")
+        .hasMatch(text);
```
Comment (ShanghaiG) [inline frontend/lib/core/service/validation_service.dart:9]: Could be one space between these two lines

File: frontend/lib/screens/common_widgets/text_field.dart [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: double.infinity,
+      padding: const EdgeInsets.symmetric(horizontal: 20),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          _createHeader(),
```
Comment (ShanghaiG) [inline frontend/lib/screens/common_widgets/text_field.dart:72]: nice, thats better approach for me

**Review by ShanghaiG — APPROVED** [review]

File: frontend/lib/core/service/validation_service.dart (hunk 3 above)
Comment (jowczar) [reply frontend/lib/core/service/validation_service.dart:8]: it's copied from tutorial
normally you'd probably want to use some package for this and form validation
not sure if we need one just for this small n simple uni app

File: frontend/lib/core/service/validation_service.dart (hunk 4 above)
Comment (jowczar) [reply frontend/lib/core/service/validation_service.dart:3]: probably, but that'd be hacky and unnecessary shadowing code
here you can see the number that is expected and can change it easily

**Pull request merged**